flaskApi/app.py

```diff
@@ -0,0 +1,77 @@
+import logging
+from zipfile import BadZipFile
+from flask import Flask, request, jsonify
+import app_service
+
+app = Flask(__name__)
+
+# Configure the logger
+logging.basicConfig(level=logging.DEBUG)
+
+
+@app.route('/upload', methods=['POST'])
+def upload_file():
+    # Log request details
+    log_message = f"Request Details: Method={request.method}, URL={request.url}, IP={request.remote_addr}, User-Agent={request.user_agent}"
+    logging.info(log_message)
+
+    # check if file send
+    if 'file' not in request.files:
+        error_msg = 'No file found in request!'
+        logging.error(error_msg)
+        return jsonify({'error_msg': error_msg}), 400
+
+    file = request.files['file']
+
+    # Check if a file is selected
+    if file.filename == '':
+        error_msg = 'No file selected!'
+        logging.error(error_msg)
+        return jsonify({'error_msg': error_msg}), 400
+
+    try:
+        uid = app_service.upload_pptx(file)
+    except (BadZipFile, ValueError) as e:
```

Comment:
This name does not adhere to python's naming conventions. Names should be indicative, and when used properly can reduce redundant documentation. In addition, each Variable, function, module, constant and class names has a specific naming convention. Read this link to learn the proper conventions - https://visualgit.readthedocs.io/en/latest/pages/naming_convention.html. If you don't know what CamelCase and snake_case means, read this link - https://www.google.com/amp/s/www.theserverside.com/answer/Camel-case-vs-snake-case-Whats-the-difference%3famp=1 . For a general overview on python conventions (PEP-8), read this link - https://peps.python.org/pep-0008